c_src/bin_nif.c

```diff
@@ -89,6 +90,22 @@ from_list(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
 }
 
 static ERL_NIF_TERM
+length_r(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
```

Comment:
You are right. I made it C not for speed, but to be able to to use C level sizeof macro that can calculate memory taken by underlying structure. That can be different on different architectures due to alignment.